qoqo-iqm/src/devices/demo_device.rs

```diff
@@ -69,9 +69,9 @@ impl DemoDeviceWrapper {
 impl DemoDeviceWrapper {
     /// Create new simulator device.
     #[new]
-    pub fn new() -> Self {
+    pub fn new(endpoint_url: String, number_qubits: usize) -> Self {
         Self {
-            internal: DemoDevice::new(),
+            internal: DemoDevice::new(endpoint_url, number_qubits),
```

Comment:
I should have been clearer in specifying this. I would not have the endpoint_url as a parameter of new but in new always set the default one and have an extra function `set_endpoint_url` to change it.